src/huggingface_hub/repository.py

```diff
@@ -432,3 +441,42 @@ def push_to_hub(self, commit_message="commit files to HF hub") -> str:
         self.git_add()
         self.git_commit(commit_message)
         return self.git_push()
+
+    @contextmanager
+    def commit(
+        self,
+        commit_message: str,
+    ):
+        """
+        Context manager utility to handle committing to a repository.
```

Comment:
nit: I would add an example here

Reply:
should there be a warning/error if we're not using this method as a context manager? (not sure whether we can detect this)

Reply:
I'm not sure about this. It would prevent some use-cases, like using it as follows:

```py
repo = Repository("text-files", clone_from="Jikiwa/text-files")

commit = repo.commit("My first file :)")

with commit:
    with open("file.txt", "w+") as f:
```

which should be a valid use case for context managers.